h/accounts/views.py

```diff
@@ -293,9 +285,11 @@ def reset_password_with_code_form(self):
         # If valid, we inject the supplied it into the form as a hidden field.
         # Otherwise, we 404.
         try:
-            user = schemas.ResetCode().deserialize(None, code)
+            user = schemas.ResetCode().deserialize(self.schema, code)
         except colander.Invalid:
             raise httpexceptions.HTTPNotFound()
+        except BadSignature:
```

Comment:
This should be caught in the the validator and reraised as `colander.Invalid`, partially because `Invalid` contains additional metadata about what was invalid that's used by `deform`, but mainly because this is exposing implementation details of the validator to its client.